Core/include/Acts/Vertexing/FullBilloirVertexFitter.ipp

```diff
@@ -294,37 +288,39 @@ Acts::FullBilloirVertexFitter<input_track_t, linearizer_t>::fit(
 
     // assign new linearization point (= new vertex position in global frame)
     linPoint += deltaV;
+
     if (newChi2 < chi2) {
       chi2 = newChi2;
 
-      Vector4 vertexPos(linPoint);
-
-      fittedVertex.setFullPosition(vertexPos);
+      fittedVertex.setFullPosition(linPoint);
       fittedVertex.setFullCovariance(covDeltaVmat);
       fittedVertex.setFitQuality(chi2, ndf);
 
       std::vector<TrackAtVertex<input_track_t>> tracksAtVertex;
 
       std::shared_ptr<PerigeeSurface> perigee =
           Surface::makeShared<PerigeeSurface>(
-              VectorHelpers::position(vertexPos));
+              VectorHelpers::position(linPoint));
+
+      for (std::size_t iTrack = 0; iTrack < billoirTracks.size(); ++iTrack) {
+        const auto& bTrack = billoirTracks[iTrack];
 
-      iTrack = 0;
-      for (auto& bTrack : billoirTracks) {
         // new refitted trackparameters
         BoundVector paramVec = BoundVector::Zero();
         paramVec[eBoundPhi] = trackMomenta[iTrack](0);
         paramVec[eBoundTheta] = trackMomenta[iTrack](1);
         paramVec[eBoundQOverP] = trackMomenta[iTrack](2);
+        paramVec[eBoundTime] = linPoint[FreeIndices::eFreeTime];
```

Comment:
Don't all tracks belonging to the vertex have the same time now? Maybe just taking the time the track had before the fit would be more correct?

Reply:
right - but I guess we should use the time after fit here because I guess this is what also happens to the position

Reply:
Ah ok! So in the end we set d0=z0=0 because we say that the track passes exactly through the vertex?

Reply:
looking at this again I don't think any of this is correct :sweat_smile: but for now it is okay because this is not influencing the fit but only outputting the track parameters which are attached to the vertex. I will add a comment that this looks like it is garbage